### ISSUE TITLE
Fix Request for pmd7 AvoidUnguardedMutableInheritedFieldsInSharedObjects false positives

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#391]
If confuses local variables and method parameters with inherited objects.

```
import org.springframework.stereotype.Service;
@Service
class Issue extends Base {
    Issue(RestTemplate restTemplate) {
        restTemplate.setErrorHandler(); // <--- false positive
    }
    void bar(String text) {
      try {
       } catch (TechnicalException e) {
            e.setText(text); // <--- false positive
       }
    }
}


```

**Comment by jborgers** [comment]
local variable defined not immediately in the method
